tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+            time.sleep(5)
+            continue
+
+        temp_cases = {}
+        
+        for case in resp:
+            try:
+                temp_case = Support_Case(case)
+            except NameError as e:
+                logging.error(e, ' : {}'.format(datetime.now()))
+                loop_skip = True
+                break
+            try:
+                temp_cases[temp_case.case_number] = temp_case
+            except AttributeError as e:
+                logging.error(e, ' : {}'.format(datetime.now()))
+                loop_skip = True
+                break
+
+        if loop_skip == True:
+            time.sleep(5)
+            continue
+
+        # Remove the cases from our dictionary that are no longer active
+        for key in cases:
+            delete_entry = True
+            for temp_key in temp_cases:
+                if key == temp_key:
+                    delete_entry = False
+                    break
+            if delete_entry == True:
+                notify_slack(key, 'closed', '')
+                del cases[key]
+        
+        # Check for existing cases that have a new update time. Post their relevant update
+        # to the channels that are tracking those cases.
+        for temp_key in temp_cases:
+            if temp_key in cases:
+                if cases[temp_key]['comment_list'] != temp_cases[temp_key].comment_list:
+                    if 'googleSupport' in temp_cases[temp_key].comment_list[0]['creator']:
+                        notify_slack(temp_key, 'comment', temp_cases[temp_key].comment_list[0]['body'])
+                if cases[temp_key]['priority'] != temp_cases[temp_key].priority:
+                    notify_slack(temp_key, 'priority', temp_cases[temp_key].priority)
+                if cases[temp_key]['escalated'] != temp_cases[temp_key].escalated:
+                    if temp_cases[temp_key].escalated == True:
+                        notify_slack(temp_key, 'escalated', temp_cases[temp_key].escalated)
+                    else:
+                        notify_slack(temp_key, 'de-escalated', temp_cases[temp_key].escalated)
+        
+        # Replace the stored case list with our latest pull and update the file
+        try:
+            file_overwrite(cases_file, temp_cases)        
+        except OSError as e:
+            logging.error(e, ' : {}'.format(datetime.now()))
+            sleep_timer = 2
+
+        # Wait to try again so we don't spam the API
+        time.sleep(sleep_timer)
+
+
+def notify_slack(case, update_type, update_text):
+    """
+    Sends update messages to Slack.
+    
+    Parameters
+    ----------
+    case : str
+        unique id of the case
+    update_type : str
+        specifies what was changed in the case
+    update_text : str
+        update relevant content that is injected into the Slack message
+    """
+    if os.path.exists(tracked_cases_file):        
+        with open(tracked_cases_file) as tcf:
+            tracker = list(eval(tcf.read()))
```

Comment:
Are you sure about this eval?

Reply:
The file is generated with the `file_overwrite` function, that serializes the data using JSON. I think you should use ` json.loads` (or similar) instead of `eval`. Although you are controlling here the file, `eval` is potentially dangerous, any code that is written to that file would be executed by your program.

Reply:
I will look to change this out for something that doesn't use eval

Reply:
Fixed

Reply:
Fixed